auth/auth.go

```diff
@@ -38,5 +42,58 @@ func main() {
 }
 
 func authCreateHandler(w http.ResponseWriter, r *http.Request) {
-	json.NewEncoder(w).Encode(struct{}{})
+	var req authDAO.AuthCreateRequest
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		errMsg := utils.CreateErrorJSON(fmt.Sprintf("Invalid request parameters: %s", err.Error()))
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	_, err = valid.ValidateStruct(req)
+	if err != nil {
+		errMsg := utils.CreateErrorJSON(fmt.Sprintf("Invalid request parameters: %s", err.Error()))
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	// Hash and salt the password before storing
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(req.Password), bcrypt.DefaultCost)
+	if err != nil {
+		errMsg := utils.CreateErrorJSON(fmt.Sprintf("Could not hash password: %s", err.Error()))
+		http.Error(w, errMsg, http.StatusInternalServerError)
+		return
+	}
+
+	req.Password = string(hashedPassword)
+	err = dao.CreateAuth(req)
+	if err != nil {
+		errMsg := utils.CreateErrorJSON(fmt.Sprintf("Something went wrong: %s", err.Error()))
+		http.Error(w, errMsg, http.StatusInternalServerError)
+		return
+	}
+
+	accessToken, err := createToken(req.Email, "TODO", "TODO")
+	if err != nil {
+		errMsg := utils.CreateErrorJSON(fmt.Sprintf("Could not create access token: %s", err.Error()))
+		http.Error(w, errMsg, http.StatusInternalServerError)
+		return
+	}
+
+	response := authDAO.AuthCreateResponse{
+		AccessToken: accessToken,
+	}
+	json.NewEncoder(w).Encode(response)
+
+}
+
+// Create an access token with a 24 hour lifetime
```

Comment:
Not something to be changed for this PR, but is this something we want to be customisable?